crates/libs/core/src/param.rs

```diff
@@ -17,68 +17,10 @@ impl<T: Type<T>> Param<T> {
     }
 }
 
-#[doc(hidden)]
-pub trait TryIntoParam<T: Type<T>> {
-    fn try_into_param(self) -> Result<Param<T>>;
-}
-
-impl<T> TryIntoParam<T> for Option<&T>
-where
-    T: ComInterface,
-{
-    fn try_into_param(self) -> Result<Param<T>> {
-        match self {
-            Some(from) => Ok(Param::Borrowed(from.abi())),
-            None => Ok(Param::Borrowed(unsafe { std::mem::zeroed() })),
-        }
-    }
-}
-
-impl<T, U> TryIntoParam<T> for &U
-where
-    T: ComInterface,
-    U: ComInterface,
-    U: CanTryInto<T>,
-{
-    fn try_into_param(self) -> Result<Param<T>> {
-        if U::CAN_INTO {
-            Ok(Param::Borrowed(self.abi()))
-        } else {
-            Ok(Param::Owned(self.cast()?))
-        }
-    }
-}
-
-#[doc(hidden)]
-pub trait CanTryInto<T>: ComInterface
-where
-    T: ComInterface,
-{
-    const CAN_INTO: bool = false;
+pub trait CanInto<T>: Sized {
```

Comment:
If I understand correctly, the `CanInto` trait is used for conversions between Interfaces (through `QueryInterface` or direct) and also for reusable handles.
To avoid confusion, I think we could use different traits for these purposes, with possibly more expressive names (something like `ImplementsInterface` and `AlsoUsableForHandle` ?).

Reply:
Feel free to elaborate on a new issue. I'm not sure I want to add more traits than needed, but I'm open to ideas.